python3_user_features.rst

```diff
@@ -0,0 +1,406 @@
+it turns out that Python doesn't do anything with these type annotations (there
+are still reasons why developers might want to do this, but this is not
+necessarily critical for the typical user).
+
+However, some packages have now implemented their own annotations. For example,
+the `Astropy <http://www.astropy.org>`_ package uses these to allow users to
+specify what units different variables should be in:
+
+.. code-block:: python
+
+    >>> import astropy.units as u
+    >>> @u.quantity_input
+    ... def kinetic_energy(mass: u.kg, velocity: u.m / u.s):
+    ...    return 0.5 * mass * velocity ** 2
+
+This does then raise an error if the variables do not have units attached:
+
+.. code-block:: python
+
+    >>> kinetic_energy(1, 3)
+    ...
+    TypeError: Argument 'mass' to function 'kinetic_energy' has no 'unit'
+    attribute. You may want to pass in an Astropy Quantity instead.
+
+or if the units are not compatible/convertible:
+
+.. code-block:: python
+
+    >>> kinetic_energy(1 * u.s, 3 * u.km / u.s)
+    ...
+    UnitsError: Argument 'mass' to function 'kinetic_energy' must be in
+    units convertible to 'kg'.
+
+Other packages will hopefully also provide useful annotations such as these!
+
+Sensible comparison
+-------------------
+
+In Python 2, it was possible to compare things that shouldn't really be
+comparable:
+
+.. code-block:: python
+
+    >>> '1' > 2
+    True
+
+Whether a string was greater than an integer or a float was not necessarily
+predictable or intuitive. In Python 3, this type of comparison is no longer
+allowed:
+
+.. code-block:: python
+
+    >>> '1' > 2
+    ...
+    TypeError: unorderable types: str() > int()
+
+This should avoid quite a few bugs!
+
+String interpolation
+--------------------
+
+The upcoming Python 3.6 release will include a new type of strings: f-strings.
+The idea is that when doing string formatting, we can often end up in cases
+that are too verbose such as:
+
+.. code-block:: python
+
+    >>> value = 4 * 20
+    >>> 'The value is {value}.'.format(value=value)
+    'The value is 80.'
+
+or we can end up in situations where the code is unnecessarily complex, since
+``value`` is detached from where it appears in the string.
+
+.. code-block:: python
+
+    >>> 'The value is {}.'.format(value)
+    'The value is 80.'
+
+The new `f-strings <https://www.python.org/dev/peps/pep-0498/>`_ allow you to
+use variable names directly inside the curly brackets:
+
+.. code-block:: python
+
+    >>> f'The value is {value}.'
+    'The value is 80.'
+
+You can actually use full Python expressions inside the curly brackets! For
+instance:
+
+.. code-block:: python
+
+    >>> a, b = 10, 20
+    >>> f'The sum of the values is {a + b}.'
+    'The sum of the values is 30.'
+
+Unicode strings
+---------------
+
+In Python 2, only the basic ASCII character set was available in standard
+strings; to use the much more extensive Unicode set of characters, you had to
+prefix each string with a u:
```

Comment:
This sounds like the only benefit is that you don't need the `u` prefix in Python 3, which is really trivial.

The real benefit is harder to explain: Python 2 lets you casually mix bytes and unicode, which works when you test with `'abc'`, but will probably blow up as soon as your program meets 'Zoë'. Python 3 forces you to think about bytes and text, which is more initial effort, but gives you a better chance of having robust code. This is a tough sell for English speaking scientists who rarely need to think about fancy foreign letters, though.